tests/conftest.py

```diff
@@ -15,3 +23,45 @@ def vcr_config():
     return {
         "filter_headers": [CROSSREF_HEADER_KEY, SEMANTIC_SCHOLAR_HEADER_KEY],
     }
+
+
+@pytest.fixture()
+def unique_tmp_subfolder(tmp_path):
+    yield tmp_path
+    # Cleanup after the test
+    if tmp_path.exists():
+        shutil.rmtree(tmp_path, ignore_errors=True)
```

Comment:
By accepting `tmp_path`, this fixture is really just more like a folder cleanup and is unrelated to tempfiles. Can you rename it?

Note there is a `pytest` plugin https://github.com/pytest-dev/pyfakefs for filesystem management, and it works great I used it a lot in the past